bundles/CoreBundle/EventListener/ResponseExceptionListener.php

```diff
@@ -177,4 +187,56 @@ protected function logToHttpErrorLog(Request $request, $statusCode)
             ]);
         }
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return string
+     *
+     * @throws \Exception
+     */
+    private function determineErrorPath(Request $request): string
+    {
+        if ($this->siteResolver->isSiteRequest($request)) {
+            $path = $this->siteResolver->getSitePath($request);
+        } else {
+            $path = urldecode($request->getPathInfo());
+        }
+
+        // Find nearest document by path
+        $document = $this->documentService->getNearestDocumentByPath(
+            $path,
+            false,
+            ['page', 'snippet', 'hardlink', 'link', 'folder']
+        );
+
+        if ($document) {
+            if ($document->getProperty('language')) {
+                $locale = $document->getProperty('language');
+            }
+        }
+
+        // If locale can't be determined or localized error page for given locale is not defined
```

Comment:
We should consider the user-agent preferences as a fallback here as well (`\Symfony\Component\HttpFoundation\Request::getLanguages`)

Reply:
Adjusted as you requested.